Clarify layered deployment manifest requirements (#79777)
Azure/azure-content: articles/iot-edge/how-to-deploy-cli-at-scale.md

  }
}
```

> >[!NOTE]
> > Take note that this layered deployment manifest has a slightly different format than a standard deployment manifest. The desired properties of the runtime modules are collapsed using dot notation. This formatting is required for the Azure portal to recognize a layered deployment. For example:
> 
>   - `properties.desired.modules.<module_name>`
>   - `properties.desired.routes.<route_name>`

Reviewer: Thanks for going through a couple of iterations on these changes. I think this latest version looks great. Just one small formatting fix and then I can approve the PR. All of the lines of a "Note" box need the > symbol to format correctly, even if they're blank or indented: 

```suggestion
>[!NOTE]
> Take note that this layered deployment manifest has a slightly different format than a standard deployment manifest. The desired properties of the runtime modules are collapsed using dot notation. This formatting is required for the Azure portal to recognize a layered deployment. For example:
>
>  - `properties.desired.modules.<module_name>`
>  - `properties.desired.routes.<route_name>`
```

Author: Committed and looking forward to having it merged 😄 @kgremban

}
```

> 
> 

Author: ```suggestion
```

}
```

In summary, to create a layered deployment:

> - Must add the `--layered` flag to the Azyre CLI create command

Reviewer 2: >y [](http://example.com/codeflow?start=41&length=1)

typo

> - Must not contain any system modules, cam only user modules

Reviewer 2: >cam only user modules [](http://example.com/codeflow?start=39&length=21)

maybe reword to "it can only contain user modules"

> - Must use the 'dot notation' under `$edgeAgent` and `$edgeHub`

Reviewer 2: >Must use the 'dot notation' under `$edgeAgent` and `$edgeHub` [](http://example.com/codeflow?start=2&length=61)

The example you gave earlier was very illustrative, do you think we should add it in the docs too - properties.desired.[base] $edgeAgent.modules: { version: 2}
[layered] properties.desired.$edgeAgent.modules.version: '2'

Author: That example was for the two different ways you could write in the layered deployment, both of which are valid. Do you think it would still be good to add it?

(I've gone ahead and added the example)

}
```

The same can also be expressed with:

```json
"SimulatedTEmperatureSensor": {
  "properties.desired.layeredProperties.SendData" : true,
  "properties.desired.layeredProperties.SendInterval": 5
}
```

> > Warning: Currently there is a bug with a layered deployment that updates only properties, where the deployment is never marked as applied in the portal

Reviewer 3: > **Warning**: Currently there is a bug with a layered deployment that updates only properties.
These layers need an empty edgeAgent object (i.e.`"$edgeAgent":{}`) in the deployment manifest to be considered a valid layered deployment. 
They will also **only** shown as applied in the module twins they target. For the `$edgeAgent` module twin, and hence in the Azure portal, the layer will be marked as targeted.

> > Warning: Currently there is a bug with a layered deployment that updates only properties. These layers need an empty edgeAgent object (i.e. `"$edgeAgent":{}`) in the deployment manifest to be considered a valid layered deployment. They will also only be shown as applied in the module twins they target. For the `$edgeAgent` module twin, and hence in the Azure portal, the layer will only ever be marked as targeted, not applied.

Reviewer: A suggestion:

```suggestion
>[!NOTE]
>Currently, all layered deployments must include an edgeAgent object to be considered valid. Even if a layered deployment only updates module properties, include an empty object. For example, `"$edgeAgent":{}`. A layered deployment with an empty edgeAgent object will be shown as **targeted** in the edgeAgent module twin, not **applied**.
```

The [!NOTE] syntax will give you the nice purple box bounding this text.

In summary, to create a layered deployment:

- Must add the `--layered` flag to the Azure CLI create command
- It may not contain system modules

> - Must use the 'dot notation' under `$edgeAgent` and `$edgeHub`

Reviewer: Can you explain what you mean by "dot notation"?

Author: I've added a clarifying example now. I'm also totally open to a better word to explain the pattern 😄 
@kgremban

Reviewer: I believe that the two examples you included are just different ways of writing the same thing. You don't have to collapse the sections, it's just a shorthand. 

```
"$edgeAgent.properties.desired.modules.SimulatedTemperatureSensor": {
```

Should resolve to be the same as

```json
"$edgeAgent": {
  "properties": {
    "desired": {
      "modules": {
        "SimulatedTemperatureSensor": {
```

Have you found that's not the case?

Author: Actually, this is not the case. The non flattened format(without full dot notation) results the portal incorrectly interpreting it as a regular deployment. This then results in a "incorrect schema error" on the portal, because layered deployments don't have a schema version.

Feel free to try it out :) 

I've also spoken to Rick Kessler from the Azure team about this and they says this is intended. 

@kgremban

Author: @kgremban I've added your suggestion as discussed
